tests/integration/helpers.py

```diff
@@ -293,6 +293,25 @@ async def set_mtls_client_acls(ops_test: OpsTest, bootstrap_server: str) -> str:
     return result
 
 
+async def create_test_topic(ops_test: OpsTest, bootstrap_server: str) -> str:
+    """Creates `test` topic and adds ACLs for principal `User:*`."""
+    _ = check_output(
+        f"JUJU_MODEL={ops_test.model_full_name} juju ssh kafka/0 sudo -i 'sudo charmed-kafka.topics --bootstrap-server {bootstrap_server} --command-config {PATHS['kafka']['CONF']}/client.properties -create -topic test'",
+        stderr=PIPE,
+        shell=True,
+        universal_newlines=True,
+    )
+
+    result = check_output(
+        f"JUJU_MODEL={ops_test.model_full_name} juju ssh kafka/0 sudo -i 'sudo charmed-kafka.acls --bootstrap-server {bootstrap_server} --add --allow-principal=User:* --operation READ --operation WRITE --operation CREATE --topic test --command-config {PATHS['kafka']['CONF']}/client.properties'",
+        stderr=PIPE,
+        shell=True,
+        universal_newlines=True,
+    )
+
+    return result
+
+
```

Comment:
**suggestion**: In order to converge to a more substrate-agnostic helper, you can take some inspiration from what we have on [kafka-bundle](https://github.com/canonical/kafka-bundle/blob/main/tests/integration/e2e/helpers.py#L129-L157). There is a little more boilerplate, but then you can simply copy/paste when it's time to port to K8s

Reply:
I'd love to do that, but since other `helper` functions here are not written this way (i.e. they assume vm substrate), wouldn't that be a problem? Maybe I could do a whole refactor in a future PR?

Reply:
Yeah, no prob doing that later